packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Contracts
+import { OPContractsManager } from "src/L1/OPContractsManager.sol";
+
+// Interfaces
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { IProtocolVersions } from "interfaces/L1/IProtocolVersions.sol";
+import { IPermissionedDisputeGame } from "interfaces/dispute/IPermissionedDisputeGame.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IFaultDisputeGame } from "interfaces/dispute/IFaultDisputeGame.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+
+// Libraries
+import { Claim, GameTypes } from "src/dispute/lib/Types.sol";
+
+///  @title OPPrestateUpdater
+///  @notice A custom implementation of OPContractsManager that allows for modified deployment parameters
+contract OPPrestateUpdater is OPContractsManager {
+    /// @notice Thrown when a function from the parent (OPCM) is not implemented.
+    error NotImplemented();
+
+    /// @notice Thrown when the prestate of a permissioned disputed game is 0.
+    error PDGPrestateRequired();
+
+    // @return Version string
+    /// @custom:semver 1.0.0
+    function version() public pure override returns (string memory) {
+        return "1.0.0";
+    }
+
+    // @notice Constructs the CustomOPContractsManager contract
+    // @param _superchainConfig Address of the SuperchainConfig contract
+    // @param _protocolVersions Address of the ProtocolVersions contract
+    // @param _l1ContractsRelease Version string for L1 contracts release
+    // @param _blueprints Addresses of Blueprint contracts
+    // @param _implementations Addresses of implementation contracts
+    // @param _upgradeController Address of the upgrade controller
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        IProtocolVersions _protocolVersions,
+        IProxyAdmin _superchainProxyAdmin,
+        string memory _l1ContractsRelease,
+        Blueprints memory _blueprints,
+        Implementations memory _implementations,
+        address _upgradeController
+    )
```

Comment:
Does the prestate updater actually need this same set of inputs? Wondering if we should simplify the constuctor and deploy script to take fewer args and just pass the zero address into the OPCM constructor for the params we don't use?

Reply:
I can do that, what's your input @maurelian ?

Reply:
It's a big diff, but it's correct that we don't need all of this and it will simplify the deploy scripts a lot, so we _should_ do it.

Reply:
Okay, I will do the change

Reply:
TYVM